src/applications/check-in/hooks/usePostTravelClaims.jsx

```diff
@@ -0,0 +1,69 @@
+import { useEffect, useState, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { api } from '../api';
+import { useStorage } from './useStorage';
+import { makeSelectForm, makeSelectCurrentContext } from '../selectors';
+import { APP_NAMES } from '../utils/appConstants';
+
+const usePostTravelClaims = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [isComplete, setIsComplete] = useState(false);
+  const [travelPayClaimError, setTravelPayClaimError] = useState(false);
+  const selectForm = useMemo(makeSelectForm, []);
+  const { data } = useSelector(selectForm);
+  const { facilitiesToFile } = data;
+  const selectCurrentContext = useMemo(makeSelectCurrentContext, []);
+  const { token: uuid } = useSelector(selectCurrentContext);
+  const { setTravelPaySent, getTravelPaySent } = useStorage(
+    APP_NAMES.TRAVEL_CLAIM,
+    true,
+  );
+  const sentTravelClaims = getTravelPaySent(window);
+
+  const faciltiesToPost = facilitiesToFile.filter(
+    facility => !(facility.stationNo in sentTravelClaims),
+  );
+  useEffect(
+    () => {
+      const markTravelPayClaimSent = facilities => {
+        facilities.forEach(facility => {
+          sentTravelClaims[facility.stationNo] = new Date();
```

Comment:
Is `sentTravelClaims[]` a series of objects/array with dates?

Reply:
it is an object with station as key and date as value. But you made me realize, that I forgot to do the date eval here!